### ISSUE TITLE
The event-location is null in a right click event

### ISSUE THREAD

**Issue by TheLimeGlass** [SkriptLang/Skript#6110]
### Skript/Server Version

```markdown
2.8.0-dev
Paper 1.20.2
Java 17
```

### Bug Description

The event-location will be null in a `on right click on chest` event

### Expected Behavior

To be able to grab either the location of the clicked block, clicked entity or event-player, I have an automatic error when multiple event-values of the same type exist and are used in an ambiguous way in 2.8+. This, this just has no location associated with it.

### Steps to Reproduce

```
on right click on chest:
    message "%event-location%" # null
```

### Other

At the minimum, there should be an error stating event-location is too ambiguous, similar to using event-entity for a damage event, or get my automatic error to catch by registering an event-value relating to the location.

### Agreement

- [X] I have read the guidelines above and affirm I am following them with this report.

**Comment by CerialPvP** [comment]
tf??? where is 2.8???

**Comment by UnderscoreTud** [comment]
It's self built, not official

**Comment by TheLimeGlass** [comment]
> why is event-location ambiguous though, it'd always be the location of the clicked entity/block

That's the exact reasoning as you said. It could be either the entity location or the block location. That's ambiguity.

**Comment by moraedu** [comment]
> That's the exact reasoning as you said. It could be either the entity location or the block location. That's ambiguity.

No, because the event is only triggered for only one of those cases at the time, and event-location would always be the clicked target regardless of what they clicked. If anything, the event is ambiguous for not making it clear that it fires for both blocks and entities in its syntax however its event values (or at least this one) is not.

**Comment by TheLimeGlass** [comment]
> No, because the event is only triggered for only one of those cases at the time, and event-location would always be the clicked target regardless of what they clicked. If anything, the event is ambiguous for not making it clear that it fires for both blocks and entities in its syntax however its event values (or at least this one) is not.

Skript converts types to get the desired event values. So for example, you can right click with an item in your hand in the sky and the event-location will be the player.

I see what you mean by if they click something, it should be that clicked item, but then you're setting the presidentnce that it'll always be that clicked target, so in the case it's just a `on right click`, the event-location could be any of the three events.

I have both an event value priority pull request to address this, but most importantly in 2.8 there will be an automatic ambiguity error if we are to register the event-location, so it helps define proper definition of use location of clicked block instead of event-location. Because currently it doesn't error.

**Comment by sovdeeth** [comment]
seems to have been completed as it works fine now
